modality-probe-cli/src/export/mod.rs

```diff
@@ -0,0 +1,340 @@
+//! Export a textual representation of a causal graph using the
+//! collected columnar form as input.
+
+use std::{collections::HashMap, fmt, fmt::Write, fs::File, path::PathBuf, str::FromStr};
+
+use err_derive::Error;
+use structopt::StructOpt;
+use uuid::Uuid;
+
+use modality_probe_collector_common::{json, Error as CollectorError};
+
+use crate::{component::Component, events::Events};
+
+mod graph;
+use graph::{EventMeta, ProbeMeta};
+
+/// Export a textual representation of a causal graph using the
+/// collected coumnar form as input.
+#[derive(Debug, PartialEq, StructOpt)]
+pub struct Export {
+    /// Generate the graph showing only the causal relationships,
+    /// eliding the events inbetween.
+    #[structopt(short, long)]
+    pub interactions_only: bool,
+    /// The path to a component directory. To include multiple
+    /// components, provide this switch multiple times.
+    #[structopt(short, long, required = true)]
+    pub components: Vec<PathBuf>,
+    /// The path to the collected trace.
+    #[structopt(short, long, required = true)]
+    pub report: PathBuf,
+    /// The type of graph to output.
+    ///
+    /// This can be either `cyclic` or `acyclic`.
+    ///
+    /// * A cyclic graph is one which shows the possible paths from
+    /// either an event or a probe.
+    ///
+    /// * An acyclic graph shows the causal history of either all
+    /// events or the interactions between traces in the system.
+    #[structopt(required = true)]
+    pub graph_type: GraphType,
```

Comment:
I don't know if CLI-API is up for review/discussion, but I am somewhat surprised by the cyclic/acyclic terminology here.  In practice, this flag seems to control the difference between producing a causal history or a summarized/collapsed version of the state-transitions that occurred in such a history.

Reply:
Yeah, those were @nathanaschbacher's too. Again, I don't have a strong opinion on what to call these. The "cyclic" ones are event or probe oriented while the "acyclic" ones are based on causality.

Reply:
Marking this one as resolved for now, as we're not sure what to permanently call this yet.